src/magic_enchantment.cpp

```diff
@@ -181,6 +181,31 @@ bool enchantment::is_active( const Character &guy, const item &parent ) const
     return false;
 }
 
+// TODO: add move mutation ref and checks here
+bool enchantment::is_active(const Character &guy) const
+{    
+
+    if (active_conditions.second == condition::ALWAYS) {
+        return true;
+    }
+
+    // has no items so item-conditions will fail instantly
+    if (active_conditions.first == has::HELD ||
```

Comment:
i don't think this if statement is totally necessary in this particular function, especially since a goal here would be encapsulating the functions a bit for reuse. We can make another overload (and leave this one) that is `bool enchantment::is_active( const Character &guy, const mut_id &mid ) const` with this little bit in it.

Reply:
made changes accordingly